tests/unit/streams/adapters/EventedStreamSource.ts

```diff
@@ -0,0 +1,76 @@
+import * as registerSuite from 'intern!object';
+import * as assert from 'intern/chai!assert';
+
+import Evented from 'src/Evented';
+import ReadableStream from 'src/streams/ReadableStream';
+import ReadableStreamReader, { ReadResult } from 'src/streams/ReadableStreamReader';
+import EventedStreamSource from 'src/streams/adapters/EventedStreamSource';
+
+let emitter: Evented;
+let stream: ReadableStream<Event>;
+let source: EventedStreamSource;
+let reader: ReadableStreamReader<Event>;
+let testEvent = {
+	type: 'testEvent',
+	test: 'value'
+};
+
+registerSuite({
+	name: 'EventedStreamSource',
+
+	beforeEach() {
+		emitter = new Evented();
+		source = new EventedStreamSource(emitter, 'testEvent');
+		stream = new ReadableStream<Event>(source);
+		reader = stream.getReader();
+
+		return stream.started;
+	},
+
+	start() {
+		emitter.emit(testEvent);
+
+		return reader.read().then(function (result: ReadResult<Event>) {
+			assert.strictEqual(result.value, testEvent,
+				'Event read from stream should be the same as the event emitted by emitter');
+		});
+	},
+
+	'event array'() {
+		let appleEvent = {
+			type: 'apple',
+			test: 'value'
+		};
+		let orangeEvent = {
+			type: 'orange',
+			test: 'value'
+		};
+
+		source = new EventedStreamSource(emitter, [ 'apple', 'orange' ]);
```

Comment:
It was done like this in anticipation of having more tests that used the same value for source... but in the current suite, it is indeed unnecessary - there are only two test cases that use source, and they each use a different value.